docs/pages/guides/keyboard-handling.mdx

```diff
@@ -0,0 +1,262 @@
+
+To dismiss the keyboard, use the `dismiss` method.
+
+```tsx DismissKeyboardButton.tsx
+import { Keyboard, Button } from 'react-native';
+
+export default function DismissKeyboardButton() {
+  return <Button title="Dismiss keyboard" onPress={Keyboard.dismiss} />;
+}
+```
+
+### Keyboard avoiding view
+
+The `KeyboardAvoidingView` is a component that automatically adjusts its height, position, or bottom padding based on the keyboard height to remain visible while the keyboard is displayed.
+
+```tsx HomeScreen.tsx
+import { KeyboardAvoidingView, TextInput } from 'react-native';
+
+export default function HomeScreen() {
+  return (
+    <KeyboardAvoidingView behavior={Platform.OS === 'ios' ? 'padding' : undefined} style={{ flex: 1 }}>
+      <TextInput placeholder="Type here..." />
+    </KeyboardAvoidingView>;
+  );
+}
+```
+
+In the above example, the height of the `KeyboardAvoidingView` automatically adjusts based on the device's keyboard height, which ensures that the input is always visible.
+
+When using a Bottom Tab navigator on Android, you might notice that focusing on an input field causes the bottom tabs to be pushed above the keyboard. To address this issue, add the `softwareKeyboardLayoutMode` property to your Android configuration in [app confg](/workflow/configuration/) and set it to `pan`.
+
+```json app.json
+"expo" {
+  "android": {
+    "softwareKeyboardLayoutMode": "pan"
+  }
+}
+```
+
+After adding this property, restart your app so that the changes take effect.
+
+## Advanced keyboard handling
+
+For more complex keyboard interactions, consider using the [Keyboard Controller](https://kirillzyusko.github.io/react-native-keyboard-controller) library. It offers additional functionality beyond the built-in React Native keyboard APIs, providing consistency across Android and iOS with minimal configuration and offering the native feel users expect.
+
+### Install keyboard controller
+
+To get started with [Keyboard Controller](https://kirillzyusko.github.io/react-native-keyboard-controller), install it in your Expo project:
+
+<Terminal cmd={['$ npx expo install react-native-keyboard-controller']} />
+
+Keyboard Controller uses native code, so you need to prebuild your app:
+
+<Terminal cmd={['$ npx expo prebuild --clean']} />
+
+Keyboard Controller library also requires `react-native-reanimated` to work properly. Follow the [installation instructions](/versions/latest/sdk/reanimated/#installation) to install it.
+
+### Set up Keyboard Controller provider
+
+To finalize the setup, add the `KeyboardProvider` to your app.
+
+```tsx _layout.tsx
+import { Stack } from 'expo-router';
+import { KeyboardProvider } from 'react-native-keyboard-controller';
+
+export default function RootLayout() {
+  return (
+    <KeyboardProvider>
+      <Stack>
+        <Stack.Screen name="basic" />
+        <Stack.Screen name="advanced" />
+        <Stack.Screen name="+not-found" />
+      </Stack>
+    </KeyboardProvider>
+  );
+}
+```
+
+### Usage
```

Comment:
As I recall, you don't have to use it like this, right? Is this needed for simply accounting for keyboard space, or is it only needed when animating something else along with the keyboard, where frame-tracking is needed? 

If we include this, we should ground it in why you would want to do this instead of using `KeyboardAvoidingView` as opposed to how it is similar, because this is more complex than using `KeyboardAvoidingView`.

Reply:
UPDATE: seeing where this fits in with the video demonstration, makes more sense. I'd still put this below the `KeyboardAwareScrollView` part, as that covers a very common `KeyboardAvoidingView` pain point that's very plug-and-play fixable. 

So yeah, I'd say change the heading to indicate what we're specifically trying to do (e.g., smoothly animate the view with the keyboard height) and introduce this section with the scenario described in the video so readers understand why you might want to do this.

Reply:
The provider is required! for these reasons: https://kirillzyusko.github.io/react-native-keyboard-controller/docs/recipes/platform-differences

Reply:
yeah, I understand that the provider is required, I was just thinking that you don't always need to hook in with an animation worklet, that just wrapping with the provider opens the door to using something more basic like the `KeyboardAwareScrollView`.

Reply:
Oh, I understand what you're saying. Yes, it is possible to use `KeyboardAwareScrollView` (I just tested it, and it works fine with the chat screen). The downside is that if you use a `FlatList,` you'll get the `VirtualizedList` warning (see screenshot).

<img width="435" alt="image" src="https://github.com/user-attachments/assets/efbbe17b-632a-479c-8990-5149ad9c1e07">

We can mention this as a more accessible alternative solution, but for more extensive lists, a `ScrollView` won't be very performant. Using an animated style allows for extra customization and IMO; it feels more snappy.